Add E2E tests to ensure non-admin users can log to wp-admin with no errors

allie500 opened the pull request:
Fixes #5101 

#### Changes proposed in this Pull Request

<!--
Title: A descriptive, yet concise, title.
-->

* This PR adds a new `editor` user to the E2E testing environment. The new `editor` user is created during the E2E environment setup.
* This PR updates the `auth.setup.ts` to include logging in the editor.
* New functions are added to the `helpers.ts` file to login the editor and get the editor. It also adds a storage file for the editor's login data.
* A new test spec has been added which tests that the editor (non-admin) user can successfully login and view `wp-admin` pages. One test is for a site that is fully onboarded to WooPayments. The second site covers the editor being able to login and view `wp-admin` pages before the site is onboarded and then after the site is onboarded. These tests follow the pattern outlined in the [Open admin area as non-admin](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows#open-admin-area-as-non-admin) critical flow test. This test spec is also tagged as critical.
* Lastly, I cleaned up the `default.json` file. Since Puppeteer has been removed, the only data we need in the file is the user data. I removed all the other data and renamed it to `uesrs.json`. I updated the reference to it in `shared.sh` and updated the variable it was assigned to. The variable is used in `setup.sh` so I updated variable references there as well.

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Make sure your e2e environment is down and cleaned up: `npm run test:e2e-reset`.
* Check out this branch.
* Bring up your e2e environment: `npm run test:e2e-setup`
* Once the environment is up, login to your e2e client site as the admin and navigate to users. Verify that you see the `editor` user.
* Run the test spec: `npm run test:e2e non-admin-wp-admin-access.spec`
* Confirm that the tests pass.
* GitHub action tests should be passing as well. 

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

allie500 committed:
Initial commit

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10374` or branch name `dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 2a4e507abeaeae6de5a69d1f43616ccfe340d3f4
- Build time: 2025-02-18 18:36:03 UTC

_Note: the build is updated when a new commit is pushed to this PR._

allie500 committed:
Update default configs

allie500 committed:
Update setup script

allie500 committed:
Add editor login functionality

allie500 committed:
Add new test spec

allie500 committed:
Add changelog

allie500 committed:
Update console.log string

allie500 committed:
Rename default.json to users.json
Remove everything from users.json except user data.
All the other data is no longer needed.

allie500 committed:
Update variable name and value for new users.json

allie500 committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

allie500 committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

allie500 committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

allie500 committed:
Test timeline 403 and 500 fix

allie500 committed:
Revert "Test timeline 403 and 500 fix"

This reverts commit f5516fd63f7b9fec9c683f22ce13cf506fcce6eb.

allie500 committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

eduardoumpierre committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

allie500 committed:
Merge branch 'develop' into dev/5101-add-e2e-test-non-admin-users-can-login-to-wp-admin

eduardoumpierre reviewed (approved):
The E2E tests are passing and the code changes look good. Good job on removing the `default.json` file! :shipit:
